distributed/pubsub.py

```diff
@@ -3,20 +3,25 @@
 from contextlib import suppress
 import logging
 import threading
+import typing
 import weakref
 
 from .core import CommClosedError
 from .metrics import time
 from .utils import sync, TimeoutError, parse_timedelta
 from .protocol.serialize import to_serialize
 
+if typing.TYPE_CHECKING:
```

Comment:
I found adding type annotations to the `stream_handlers` object was useful in implementing this (cf #2803), allowing type checkers to catch which handlers were async or sync. *However*, I needed this pretty unsightly hack to avoid a circular import, just to get the type name in the module scope. I'm not happy about it, and it could be removed. On the other hand, it's kind of nice for refactoring.

Reply:
Does `from __future__ import annotations` help at all here? Admittedly that is Python 3.7+ only. Though we are planning to drop Python 3.6 soon ( https://github.com/dask/distributed/pull/4390 )

Reply:
Yeah, I had to remove `annotations` in 4f1944b for 3.6 support. Though it didn't help that much, it mostly meant I didn't have to quote the type names below. The circular import problem remained.

Reply:
@jakirkham it doesn't help - you still need those imports. ``from __future__ import annotations`` helps when you have a class accepting itself or returning itself in its methods, or when class A in a module accepts or returns class B which is declared afterwards in the same module.